presenters/trial.js

```diff
@@ -58,6 +58,17 @@ function decorateDocuments(documents) {
   return documentsByCategory;
 }
 
+function decorateFDADocuments(documents) {
+  const docs = _.filter(documents, (doc) => doc.source_id === 'fda');
+  _.map(docs, (doc) => {
+    if (doc.fda_approval) {
+      // eslint-disable-next-line no-param-reassign
+      doc.application_id = doc.fda_approval.fda_application.id;
+    }
+  });
```

Comment:
It feels like you're using `map` here as if it was a `forEach`. The return value of a `map` is the initial array with the transformation of running its callback on each of its items. Here, you're instead altering the actual `doc` objects by adding an attribute to it. As they're objects (and not "primitives" like small integers), the `docs` array contain pointers to their variables, and not the variables themselves. This way, on `map`, you're changing the original models and not returning anything, so the result of it will be `[undefined, undefined, undefined, ...]` (as you're not returning anything).

That doesn't matter, though, because you're not returning the results of the map, but the original array (with items now changed).

You could change this code to either:

```javascript
const fdaDocs = _.filter(documents, { source_id: 'fda' })
return fdaDocs.map((doc) => {
  if (doc.fda_approval) {
    // eslint-disable-next-line no-param-reassign
    doc.application_id = doc.fda_approval.fda_application.id;
  }
  return doc;
});
```

Or even more "functionalish":

```javascript
return _.filter(documents, { source_id: 'fda' })
  .map((doc) => {
    if (doc.fda_approval) {
      // eslint-disable-next-line no-param-reassign
      doc.application_id = doc.fda_approval.fda_application.id;
    }
    return doc;
  });
```